### PR TITLE
Use and report to codeclimate

### LINKED ISSUE #24: Reimplement a code coverage strategy
# Reimplement a code coverage strategy

## Summary

Followup of https://github.com/linterjs/core/pull/22#issuecomment-663225211

## Motivation

Code coverage can help to provide stable and high-quality code.

## Detailed Design

Decide a code coverage strategy and implement it. GitHub Actions CI should run this at each `push`

## Documentation

Maybe not needed

## Drawbacks

None.

## Alternatives

None.

## Open Questions

Any questions?

## Help Needed

Yes, we should create the file together.

## Frequently Asked Questions

Currently none

## Related Discussions

Currently none

### PR DESCRIPTION
**Pull request by Shinigami92**
closes #24

### CODE REVIEW

**Comment by Shinigami92** [comment]
@zimme We need to provide a `CC_TEST_REPORTER_ID` secret: https://github.com/linterjs/core/runs/1003118435?check_suite_focus=true#step:10:48

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,7 +36,7 @@ jobs:
         run: yarn outdated
         continue-on-error: true
```
Comment (Shinigami92) [inline .github/workflows/ci.yml:37]: It seems that this is not only configured for this one step, but for every other below.
Like a switch case fallthrough?! 🤔

Comment (zimme) [reply .github/workflows/ci.yml:37]: I have no experience with GitHub Actions yet so I don't know tbh. I'm thinking that we can remove `yarn outdated` from the ci flow and maybe check on dependabot or something like that if we can configure it to only create PR's for MAJOR and possibly MINOR version and not for PATCH versions as it seems to do by default which is just super annoying.

**Comment by zimme** [comment]
I added the `CC_TEST_REPORTER_ID` to this repo's secrets and triggered a re-run of the jobs for this PR to see if it works.

You can find the id when you login in to code climate and chose this repo under open-source projects and go to settings and test coverage.

Feel free to see if you have access to the code climate project, but I do believe you should have becuase they use GitHub authentication.

File: README.md [hunk 2]
```diff
@@ -1,11 +1,9 @@
 # @linter/core
 
-<!-- [![codecov](https://codecov.io/gh/linterjs/core/branch/master/graph/badge.svg)](https://codecov.io/gh/linterjs/core) -->
-<!-- [![Maintainability](https://api.codeclimate.com/v1/badges/747c6f23ee4dfa61f522/maintainability)](https://codeclimate.com/github/linterjs/core/maintainability) -->
```
Comment (zimme) [inline README.md:4]: Maybe the maintainability should be left here as that's from code climate?

**Review by zimme — APPROVED** [review]
Check on that error thing, other than things look good.

**Review by zimme — APPROVED** [review]

**Pull request merged**